src/views/Workflow.vue

```diff
@@ -0,0 +1,194 @@
+<template>
+  <div id="workflow-panel" class="fill-height">
+    <toolbar />
+    <workflow :workflow-tree="workflowTree" />
+  </div>
+</template>
+
+<script>
+import { mixin } from '@/mixins/index'
+import { mapState } from 'vuex'
+import Toolbar from '@/components/cylc/Toolbar'
+import Workflow from '@/components/cylc/workflow/Workflow'
+import { convertGraphQLWorkflowToTree } from '@/components/cylc/tree/index'
+
+// query to retrieve all workflows
+const QUERIES = {
+  root: `
+    {
+      workflows(ids: ["WORKFLOW_ID"]) {
+        id
+        name
+        status
+        owner
+        host
+        port
+        taskProxies(sort: { keys: ["cyclePoint"] }) {
+          id
+          state
+          cyclePoint
+          latestMessage
+          firstParent {
+            id
+            name
+            cyclePoint
+            state
+          }
+          task {
+            meanElapsedTime
+            name
+          }
+          jobs(sort: { keys: ["submit_num"], reverse:true }) {
+            id
+            batchSysName
+            batchSysJobId
+            host
+            startedTime
+            submittedTime
+            finishedTime
+            state
+            submitNum
+          }
+        }
+        familyProxies (sort: { keys: ["firstParent"]}) {
+          id
+          name
+          state
+          cyclePoint
+          firstParent {
+            id
+            name
+            cyclePoint
+            state
+          }
+        }
+        nodesEdges {
+          nodes {
+            id
+            label: id
+            parent: firstParent {
+              id
+              state
+            }
+            state
+            cyclePoint
+            task {
+              name
+            }
+            jobs(sort: {keys: ["submit_num"], reverse: true}) {
+              id
+              batchSysName
+              batchSysJobId
+              host
+              startedTime
+              submittedTime
+              finishedTime
+              state
+              submitNum
+            }
+          }
+          edges {
+            id
+            source
+            target
+            label: id
+          }
+        }
+      }
+    }
+  `
+}
+
+export default {
+  mixins: [mixin],
+  name: 'Workflow',
+  props: {
+    workflowName: {
+      type: String,
+      required: true
+    }
+  },
+  components: {
+    toolbar: Toolbar,
+    workflow: Workflow
+  },
+  metaInfo () {
+    return {
+      title: this.getPageTitle('App.workflow', { name: this.workflowName })
+    }
+  },
+  data: () => ({
+    viewID: '',
+    subscriptions: {},
+    isLoading: true
+  }),
+  computed: {
+    ...mapState('workflows', ['workflows']),
+    currentWorkflow: function () {
+      for (const workflow of this.workflows) {
+        if (workflow.name === this.workflowName) {
+          return workflow
+        }
+      }
+      return null
+    },
+    workflowTree: function () {
```

Comment:
I think this interface kinda breaks the idea of views being able to request the data they subscribe to as the workflow component is subscribing to a superset of everything even if it is not needed (e.g. only job statuses/submit numbers required for the graph view).

Could we move this functionality into the store?

Reply:
I think you are right. Will try to move to Vuex as a `getter` :+1:

Reply:
The `currentWorkflow` was moved to the store recently too (above the code from this comment). Replacing both by `mapGetters`.

Reply:
Done

Reply:
Wow, that was quick!